### PR TITLE
Add axios

### PR DESCRIPTION
**Pull request by 7091lapS**
- Add [axios](https://github.com/mzabriskie/axios) library for managing api calls
- Api calls to actual back end data
- Add basic error display
- Get rid of useless filters
- Refactor how filters send the payload
screen shot 2017-08-10 at 11 55 45

### CODE REVIEW

File: public/js/services/Api.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import axios from 'axios';
+import moment from 'moment';
+
+export const httpClient = axios.create({
+    // baseURL: 'https://productionmetrics.local.dev-gutools.co.uk/', // Use this url in dev. TODO: Add proper ENV management
```
Comment (susiecoleman) [inline public/js/services/Api.js]: We could handle this by sending a client config object through to frontend. This is something the targeting app does https://github.com/guardian/targeting/blob/master/app/controllers/Application.scala#L19 but I think this can be left for another PR

File: public/styles/components/_chart.scss [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 
 $loadingLabelWidth: 100px;
-$loadingLabelHeight: 30px;
+$loadingLabelHeight: 300px;
 
 .chart-wrap:after {
   content: 'Loading\2026'; // Ellipsis
   display: none;
-  @include center-absolutely(100px, 30px); // Center the chart wrap passing in the width and height
+  @include center-absolutely($loadingLabelWidth, $loadingLabelHeight); // Center the chart wrap passing in the width and height
```
Comment (susiecoleman) [inline public/styles/components/_chart.scss]: Nice 👍

File: public/js/services/Api.js (hunk 1 above)
Comment (7091lapS) [inline public/js/services/Api.js]: yep, will be in a separate pr! 👍

File: public/js/components/Charts/AreaChartWrap.js [hunk 3]
```diff
@@ -4,9 +4,9 @@ import ChartTheme from '../ChartTheme/theme';
 
 const customisedTheme = Object.assign({}, Themes.simple, ChartTheme);
 
-const AreaChartWrap = ({ title, series, xLabel, yLabel, isUpdating, scale, stacked }) => {
+const AreaChartWrap = ({ title, series, xLabel, yLabel, isUpdating, scale, stacked, errors }) => {
     return (
-        <div className={`chart-wrap ${isUpdating && 'chart-wrap__updating'}`}>
+        <div className={errors && !errors.hasErrors ? `chart-wrap ${isUpdating && 'chart-wrap__updating'}` : 'error'}>
```
Comment (gtrufitt) [inline public/js/components/Charts/AreaChartWrap.js]: It's worth using the BEM modifier syntax here `chart-wrap__error`

File: public/js/redux/chartsRedux.js [hunk 4]
```diff
@@ -1,28 +1,55 @@
...
 
 Effect('filterDesk', (filterObj) => {
+    Actions.resetErrors();
     Actions.updateFilter(filterObj);
     Actions.toggleIsUpdatingCharts(true);
-    getChartData(filterObj).then(chartsData => {
-        Actions.toggleIsUpdatingCharts(false);
-        Actions.updateComposerVsIncopy(chartsData);
-    });
+    api.getComposerVsIncopy(filterObj.startDate, filterObj.endDate, filterObj.desk)
```
Comment (gtrufitt) [inline public/js/redux/chartsRedux.js]: I was thinking about how we could make sure that it was easy to add charts in the future, obviously we only have one for now, but something like this might make sense to increase the ease of adding new charts. Something to think about.

```javascript
// chartList.js
export default chartList = ['composerVsIncopy']

// chartsRedux
import chartList;

chartList.map(chart => {
     api.[`get${chart}`](filterObj.startDate, filterObj.endDate, filterObj.desk)
        .then(chartData => {
            Actions.toggleIsUpdatingCharts(false);
            Actions[`update${chart}`](chartData);
        })
        .catch(error => {
            Actions.toggleIsUpdatingCharts(false);
            Actions.[`get${chart}Failed`](error);
        });
})
```

File: public/js/redux/chartsRedux.js [hunk 5]
```diff
@@ -1,28 +1,55 @@
...
+        .then(composerVsInCopyData => {
+            Actions.toggleIsUpdatingCharts(false);
```
Comment (gtrufitt) [inline public/js/redux/chartsRedux.js]: We chatted a bit about this yesterday - Had a bit of a thought about a good candidate for a hook:

```javascript
const updatingChartHook = Hook(action => {
// Not exactly the most solid of checks, and may need 'includes' polyfill
  if (action.type.includes('update') || action.type.includes('Failed') ) { 
    Actions.toggleIsUpdatingCharts(false)
  }
})
```

Then the other bit get cleaner:

```javascript
api[`get${chart}`](filterObj.startDate, filterObj.endDate, filterObj.desk)
        .then(Actions[`update${chart}`])
        .catch(Actions[`get${chart}Failed`]);
```

File: public/js/components/Charts/AreaChartWrap.js (hunk 3 above)
Comment (7091lapS) [inline public/js/components/Charts/AreaChartWrap.js]: Yep, better! No `return` in the else line though, or that will always return isUpdating!
